File: 0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] height) {
        int l=0,r=height.length-1;
        int max=0;
        while(l<r){
            int cur=Math.min(height[l],height[r])*(r-l);
            max=Math.max(max,cur);
            if(height[l]<height[r]){
                l++;
            }else{
                r--;
            }
        }
        return max;
    }
}

File: 0011-container-with-most-water/solution.c
int maxArea(int* height, int n) {
    int area=0;
    int l=0,r=n-1;
    while(l<r){
        int min=INT_MAX;
        int b=r-l;
        if(height[l]<height[r]){
            min=height[l];
            l++;
        }else{
            min=height[r];
            r--;
        }
        int cur = min * b;
        if(area<cur){
            area=cur;
        }
    }
    return area;
}
